web/src/routes/~_dashboard/~file/ReportIssueButton.tsx

```diff
@@ -66,142 +64,133 @@ interface ReportIssueButtonProps {
   context: UserIssueContext;
 }
 
-export function ReportIssueButton({
+export const ReportIssueButton = ({
   className,
   context,
-}: ReportIssueButtonProps) {
+}: ReportIssueButtonProps) => {
   const openState = useOpenState(false);
-  const defaultValues = {
-    issueSelection: new Array(issueStrings.length).fill(false),
-    isOtherSelected: false,
-    userInputedIssue: "",
-  };
 
   const form = useForm({
     formSchema,
-    defaultValues: defaultValues,
+    defaultValues: {
+      issueSelection: issueStrings.map(() => false),
+      isOtherSelected: false,
+      userInputedIssue: "",
+    },
   });
-  const [otherExpanded, setOtherExpanded] = useState(false);
+
+  const handleClose = () => {
+    openState.close();
+    form.reset();
+  };
 
   const handleSubmit = form.handleSubmit(async (medicalReport) => {
     const selectedIssueStrings = medicalReport.issueSelection
       .map((isUserSelected, index) =>
-        isUserSelected ? issueStrings[index] : null,
+        isUserSelected ? issueStrings.at(index) : null,
       )
-      .filter((val) => val !== null);
+      .filter(Boolean);
 
-    if (
-      selectedIssueStrings ||
-      (medicalReport.isOtherSelected && medicalReport.userInputedIssue)
-    ) {
-      let issueContent = {};
-      if (selectedIssueStrings) {
-        issueContent = {
-          ...issueContent,
-          pre_defined_issues: selectedIssueStrings,
-        };
-      }
-      if (medicalReport.isOtherSelected) {
-        issueContent = {
-          ...issueContent,
-          user_inputed_issue: medicalReport.userInputedIssue,
-        };
-      }
-
-      const collectionReference = collection(
-        firestore,
-        "users_reported_issues",
+    try {
+      await addDoc(collectionRefs.usersReportedIssues(), {
+        pre_defined_issues:
+          selectedIssueStrings.length ? selectedIssueStrings : null,
+        user_inputed_issue:
+          medicalReport.userInputedIssue ?
+            medicalReport.userInputedIssue
+          : null,
```

Comment:
I would go with the one below since a user could enter something and then disable again

```suggestion
        user_inputed_issue:
          medicalReport.isOtherSelected ?
            medicalReport.userInputedIssue
          : null,
```

Reply:
Enabling/Disabling "Other" based on the value of the field is a great idea!